runelite-client/src/test/java/net/runelite/client/plugins/opponentinfo/OpponentInfoPluginTest.java

```diff
@@ -0,0 +1,199 @@
+
+        // some npc attacks the player
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(aggroFirst, thePlayer));
+
+        // verify that the current opponent is the aggressive npc
+        assertSame(opponentInfoPlugin.getLastOpponent(), aggroFirst);
+
+        // the player is attacked by a different npc
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(aggroRecent, thePlayer));
+
+        // verify that the current opponent is the most recent aggressor
+        assertSame(opponentInfoPlugin.getLastOpponent(), aggroRecent);
+
+        // an npc that is not the current opponent targets another player
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(aggroFirst, otherPlayer));
+
+        // verify that the current opponent is still the most recent aggressor
+        assertSame(opponentInfoPlugin.getLastOpponent(), aggroRecent);
+
+        // the current opponent switches targets to another player
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(aggroRecent, otherPlayer));
+
+        // verify that there is no longer an opponent
+        // (if lastTime is not null, then the lastOpponent will become null after OpponentInfoPlugin.WAIT time)
+        assertTrue(opponentInfoPlugin.getLastTime() != null || opponentInfoPlugin.getLastOpponent() == null);
+    }
+
+    /*
+     * Verify that the current opponent is the expected npc in singles
+     */
+    @Test
+    public void testSingles() {
+        Player thePlayer = mock(Player.class), otherPlayer = mock(Player.class);
+        NPC playerNPC = mock(NPC.class);
+        when(client.getLocalPlayer()).thenReturn(thePlayer);
+
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(thePlayer, null));
+
+        // verify that there is currently no opponent
+        assertNull(opponentInfoPlugin.getLastOpponent());
+
+        // some npc attacks the player
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(playerNPC, thePlayer));
+
+        // verify that the attacking npc is the current opponent
+        assertSame(opponentInfoPlugin.getLastOpponent(), playerNPC);
+
+        // the npc stops attacking the player
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(playerNPC, null));
+
+        // verify that there is no longer an opponent
+        // (if lastTime is not null, then the lastOpponent will become null after OpponentInfoPlugin.WAIT time)
+        assertTrue(opponentInfoPlugin.getLastTime() != null || opponentInfoPlugin.getLastOpponent() == null);
+
+        // the player attacks the npc
+        when(thePlayer.getInteracting()).thenReturn(playerNPC);
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(thePlayer, playerNPC));
+        // the npc attacks them back
+        when(playerNPC.getInteracting()).thenReturn(thePlayer);
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(playerNPC, thePlayer));
```

Comment:
Same here, shouldn't there be another assert between these verifying `playerNPC` is now the last opponent?